artichoke-backend/src/extn/core/symbol/mod.rs

```diff
@@ -1,115 +1,14 @@
 use std::ffi::c_void;
 
 use crate::convert::{Immediate, UnboxedValueGuard};
-use crate::extn::core::array::Array;
 use crate::extn::prelude::*;
-use crate::intern::Symbol as SymbolId;
 
 pub mod ffi;
 pub mod mruby;
 pub mod trampoline;
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Symbol(SymbolId);
-
-impl Symbol {
-    pub fn all_symbols(interp: &mut Artichoke) -> Result<Array, Exception> {
-        let all_symbols = interp.all_symbols()?.map(Symbol::from).collect::<Vec<_>>();
```

Comment:
replacing the `intaglio::AllSymbols` iterator with a `'static` iterator from `spinoso-symbol` saves an allocation.